update info on Semgrep v Snyk (#1744)
semgrep/semgrep-docs: docs/faq.md

### Comparing Semgrep to Snyk

#### SAST

> Both Semgrep and Snyk offer SAST solutions that are out of the box. Semgrep makes it easier to customize the rules that run against your code. Furthermore, when you know which rules were run, you can analyze your results to see if the relevant vulnerabilities were caught.

Reviewer: ```suggestion
Both Semgrep and Snyk offer out-of-the-box SAST and SCA solutions. Semgrep makes it easier to customize the rules that run against your code. Because these rules are visible and customizable, you can analyze your results to see if the relevant vulnerabilities were caught.
```

> In addition to selecting your rules, Semgrep allows you to write custom rules to capture use cases driven by your organization's goals. To help you write rules, Semgrep Editor provides a structure mode to guide you through the process, allows you to test your in-progress rules, and adds them to your organization’s [Policies page](/semgrep-code/policies). Semgrep offers rule-writing capabilities to all users, while Snyk limits it to Enterprise users.

Reviewer: ```suggestion
In addition to selecting your rules, Semgrep allows you to write custom rules to capture use cases driven by your organization's goals. To help you write rules, [Semgrep Editor](https://semgrep.dev/playground) provides a structure mode to guide you through the process, allows you to test your in-progress rules, and adds them to your organization’s [Policies page](/semgrep-code/policies). Semgrep offers rule-writing capabilities to all users, while Snyk limits it to Enterprise users.
```

> Both Semgrep and Snyk offer remediation advice for findings identified during scans. Snyk displays its recommendations in its UI and CLI, while Semgrep displays remediation advice and guidance in its UI and in the form of PR or MR comments.

Reviewer: Semgrep also displays remediation advice in the CLI, even the human-written autofix is in the CLI. We're more than at parity with Snyk in this regard.

> Both Semgrep and Snyk offer remediation advice for findings identified during scans. Snyk displays its recommendations in its UI and CLI, while Semgrep displays remediation advice and guidance in its UI, CLI, and in the form of PR or MR comments.

Reviewer: ```suggestion
Both Semgrep and Snyk offer remediation advice for findings identified during scans. Snyk displays its recommendations in its web app, in supported IDEs, and CLI, while Semgrep displays remediation advice and guidance in its web app, supported IDEs, the CLI, and in the form of PR or MR comments.
```
Just today we found out that they're [going to show AI recommendations in the IDE](https://semgrepinc.slack.com/archives/C05GY0PH79R/p1726110733326119). Because this statement is general, we can also say that we have IDE recommendations (we do, human-written autofix is available in our VSCode plugin).

Reviewer: It does seem like a pretty clunky sentence now that we have so many list items though :sweat_smile:  sorry about that.

> Snyk and Semgrep both display prioritization metrics, such as severity level, to help you decide which findings you should work on first. For SAST, Snyk encapsulates this information into a priority score, which provides you with information on the impact and actionability related to the finding. Semgrep, on the other hand, provides recommendations through Assistant, which offers functionality like GPT-4-powered security recommendations to help you review, triage, and remediate your Semgrep findings.

Reviewer: The paragraph starts by talking about prioritization metrics - but we compare the Snyk score with Assistant's recommendations? The two of them are not exactly the same thing - we could focus on severity/confidence/likelihood (which are all SAST metrics separate from EPSS).

Additionally, Semgrep provides action recommendations through Assistant, which offers capabilities like GPT-4-powered security recommendations to help you review, triage, and remediate your Semgrep findings.

> Snyk offers autofix capability for its SCA product, but not its SAST product. Semgrep offers autofix suggestions for SAST and SCA, where its [rules contain suggested fixes to resolve findings](/writing-rules/autofix), and in the event of a true positive where the rule doesn't have a human-written autofix, [Assistant can generate an autofix](/semgrep-assistant/overview#autofix).

Reviewer: <3 TY

Snyk offers reachability analysis for Java, JavaScript, and TypeScript, while Semgrep offers reachability analysis for [multiple languages, including Java, JavaScript, and Ruby](supported-languages/#semgrep-supply-chain)

> Snyk offers limited information on whether dependencies are direct or transitive. However, this information is only available with Enterprise plans, and the information is limited to projects using Node.js, specifically npm and Yarn packages, and Maven. Semgrep Supply Chain offers advanced reachability analysis for direct dependencies in the form of dataflow reachability. Semgrep offers this coverage for seven languages and counting.

Reviewer: ```suggestion
Snyk can detect whether dependencies are direct or transitive. However, this information is only available with Enterprise plans, and the information is limited to projects using Maven or Node.js, specifically npm and Yarn packages. Semgrep Supply Chain offers advanced reachability analysis for direct dependencies in the form of dataflow reachability. Semgrep offers this coverage for seven languages and counting.
```

Semgrep and Snyk both offer license compliance features, ensuring that the dependencies that your developers use meet the requirements set by your organization.

> To help you manage your findings, Semgrep provides you with the findings' EPSS probabilities, severity levels, transitivity information, and multiple levels of dataflow reachability. Snyk assesses impact and likelihood and encapsulates this information into a risk score. 

Reviewer: ```suggestion
To help you manage your findings, Semgrep provides you with the findings' EPSS probabilities, severity levels and transitivity information. Snyk assesses impact and likelihood and encapsulates this information into a risk score. 
```

Just removed the bit that sounded like PTT - it's not yet beta.

#### Policies and rules management 

Semgrep Code and Semgrep Secret's policies management feature provides extensive flexibility, especially with respect to a developer's workflow, by allowing results to appear:

- Only in the AppSec team’s view (monitor mode)
- In the AppSec team's view **and** in the developer’s workflow, while not failing the CI job (comment mode)
- In the AppSec team's view **and** in the developer’s workflow, while also failing the CI job (block mode)

Semgrep Supply Chain results in a failed CI job only when there are critical or high-severity findings.

#### Secrets detection

Semgrep Secrets leverages semantic analysis, entropy analysis, and validation to accurately detect and fix secrets. Snyk maintains a [business partnership with GitGuardian](https://blog.gitguardian.com/were-teaming-up-with-snyk-to-strengthen-developer-security/) to offer secrets scanning as part of Snyk Code.

> > See [Semgrep vs. Snyk](https://semgrep.dev/resources/semgrep-vs-snyk) for a detailed comparison.

Reviewer: ```suggestion
```
Not sure if this is up-to-date, I am aware that the team is working on a new Snyk page, would suggest removing the above line and then add in the new page when it's available.
